client/blocks/eligibility-warnings/index.tsx

```diff
@@ -193,8 +193,15 @@ function getSiteIsEligibleMessage(
 	}
 }
 
-function getProceedButtonText( holds: string[], translate: LocalizeProps[ 'translate' ] ) {
+function getProceedButtonText(
+	holds: string[],
+	translate: LocalizeProps[ 'translate' ],
+	isMarketplace = false
+) {
 	if ( siteRequiresUpgrade( holds ) ) {
+		if ( isMarketplace ) {
```

Comment:
`isMarketplace` is badly named here. [It's actually `isMarketplaceProduct`](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plugins/plugin-details-CTA/CTA-button.jsx#L137).

Instead, I think you can pass in the context prop and check for `context == 'plugins' || context == 'plugin-details'`.